libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -241,6 +243,33 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         navigationOptions.eHorizonOptions.minTimeDeltaBetweenUpdates
     )
 
+    // native Router Interface
+    private val nativeRouter: RouterInterface by lazy {
+        NavigatorLoader.createNativeRouterInterface(
+            navigationOptions.deviceProfile,
+            navigatorConfig,
+            createTilesConfig(
+                isFallback = false,
+                tilesVersion = navigationOptions.routingTilesOptions.tilesVersion
+            ),
+            historyRecorder.fileDirectory(),
+        )
+    }
+
+    // Router provided via @Modules, might be outer
+    private val moduleRouter: Router by lazy {
+        MapboxModuleProvider.createModule(MapboxModuleType.NavigationRouter) {
+            paramsProvider(
+                ModuleParams.NavigationRoute(
+                    accessToken
+                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ROUTER),
+                    nativeRouter,
+                    threadController
+                )
+            )
+        }
```

Comment:
Is it possible to return `null` by default? It would be easy for us to detect if a user provides custom router

Reply:
where should we return null by default? Sorry, I didn't follow

Reply:
from `MapboxModuleProvider.createModule(MapboxModuleType.NavigationRouter)`?